pkg/services/backfill.go

```diff
@@ -0,0 +1,130 @@
+	return func(ctx context.Context, intake models.SystemIntake, notes []models.Note) (bool, error) {
+		ok, err := authorize(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Printf("%+v", intake.LifecycleID)
+
+		if intake.LifecycleID.String == "" {
+			// skip it
+			return false, nil
+		}
+
+		if !ok {
+			return false, &apperrors.ResourceNotFoundError{
+				Err:      errors.New("failed to authorize backfill creation"),
+				Resource: models.Note{},
+			}
+		}
+
+		if intake.LifecycleID.ValueOrZero() == "" {
+			return false, &apperrors.BadRequestError{
+				Err: errors.New("lifecycle ID is required"),
+			}
+		}
+
+		existingRecord, existingErr := fetchIntakeByLifecycleID(ctx, intake.LifecycleID.String)
+		if existingErr != nil {
+			return false, &apperrors.QueryError{
+				Err: existingErr,
+			}
+		}
+
+		if existingRecord != nil {
+			appcontext.ZLogger(ctx).Info("updating existing system intake", zap.String("lcid", intake.LifecycleID.String))
+			intake.RequestType = models.SystemIntakeRequestTypeNEW
+			intake.Status = models.SystemIntakeStatusLCIDISSUED
+			intake.ID = existingRecord.ID
+			if _, updateErr := updateIntake(ctx, &intake); updateErr != nil {
+				return false, err
+			}
+			return false, nil
+		}
+
+		appcontext.ZLogger(ctx).Info("creating new system intake", zap.String("lcid", intake.LifecycleID.String))
+		// invariant data for all backfill (maybe do this in transport layer, for de-normalizing fields?)
+		intake.RequestType = models.SystemIntakeRequestTypeNEW // TODO: correct RequestType?
+		intake.Status = models.SystemIntakeStatusLCIDISSUED
+
+		// If no ID is provided, generate a new V3 UUID using the lifecycle ID. This will always result in
+		// the same UUID being generated when given the same input.
+		if intake.ID == uuid.Nil {
+			appcontext.ZLogger(ctx).Info("generating a new ID for lcid", zap.String("lcid", intake.LifecycleID.String))
+			intake.ID = uuid.NewMD5(uuid.NameSpaceOID, []byte(intake.LifecycleID.String))
+		}
+
+		if _, err = createIntake(ctx, &intake); err != nil {
+			return false, err
+		}
+
+		// this "mutate" section gets around the fact that LCID fields don't get saved on a CREATE operation
+		mutate, err := fetchIntake(ctx, intake.ID)
+		if err != nil {
+			return false, err
+		}
+
+		hasUpdate := false
+		if intake.LifecycleID.ValueOrZero() != "" {
+			hasUpdate = true
+			mutate.LifecycleID = intake.LifecycleID
+		}
+		if intake.LifecycleExpiresAt != nil {
+			hasUpdate = true
+			mutate.LifecycleExpiresAt = intake.LifecycleExpiresAt
+		}
+		if intake.LifecycleScope.ValueOrZero() != "" {
+			hasUpdate = true
+			mutate.LifecycleScope = intake.LifecycleScope
+		}
+		if intake.SubmittedAt != nil {
+			hasUpdate = true
+			mutate.SubmittedAt = intake.SubmittedAt
+		}
+
+		if hasUpdate {
+			if _, err = updateIntake(ctx, mutate); err != nil {
+				return false, err
+			}
+		}
+
+		for _, note := range notes {
+			// TODO: should this be done at transport layer
```

Comment:
we have an existing CreateNote service. Can we use that here?

Reply:
We are using the `store.CreateNote` method here. I don't see a `CreateNote` service anywhere, though. Maybe I'm missing it- can you point me to what you were thinking about?